### PR TITLE
Fix missing refs after 351 merge

### PR DESCRIPTION
**Pull request by valeriupredoi**
merging #351 missed a whole lot of refs  that I re-added here
A good example not to trust 100% gitHub when it tells you able to merge and do a `git merge master` in the branch before; that was an old PR that was initially pointing to `development` and the able to merge message has not been redirected towards `master` @mattiarighi can you pls approve so we can fix the `master` (currently the tests are failing there)

### CODE REVIEW

**Comment by bouweandela** [comment]
@valeriupredoi Can you please give this pull request a meaningful title that I can use in the changelog? It looks like you're fixing some incompatibility between functionality you added in #351 and #420 that got merged before  #351.

**Comment by valeriupredoi** [comment]
indeed it was a compatibility issue that should have been highlighted as conflict; Ill change it now, but also note that there is actually missing code that was not pushed to master in the test module

**Comment by valeriupredoi** [comment]
ow hang on, not the case; from #420 the add aux coord func has been removed from the test module and is actually needed by the tests I introduced

File: esmvalcore/preprocessor/_time.py [hunk 1]
```diff
@@ -199,8 +199,10 @@ def daily_statistics(cube, operator='mean'):
     iris.cube.Cube
         Daily statistics cube
     """
-    iris.coord_categorisation.add_day_of_year(cube, 'time')
-    iris.coord_categorisation.add_year(cube, 'time')
+    if not cube.coords('day_of_year'):
+        iris.coord_categorisation.add_day_of_year(cube, 'time')
+    if not cube.coords('year'):
+        iris.coord_categorisation.add_year(cube, 'time')
```
Comment (valeriupredoi) [inline esmvalcore/preprocessor/_time.py]: this is from #420 but I don't agree with the change from there!

File: tests/unit/preprocessor/_time/test_time.py [hunk 2]
```diff
@@ -762,6 +773,7 @@ def setUp(self):
             ),
             0,
         )
+        add_auxiliary_coordinate([self.cube_1, self.cube_2])
```
Comment (valeriupredoi) [inline tests/unit/preprocessor/_time/test_time.py]: the tests need aux coords for testing so I suggest reintroducing them

**Pull request merged**